src/AutoMapper/Configuration/MappingExpressionBase.cs

```diff
@@ -100,6 +113,41 @@ public void Configure(TypeMap typeMap)
             }
         }
 
+        private bool MapDestinationCtorToSource(TypeMap typeMap, ConstructorInfo destCtor, TypeDetails sourceTypeInfo, ProfileMap options)
+        {
+            var ctorParameters = destCtor.GetParameters();
+
+            if(ctorParameters.Length == 0 || !options.ConstructorMappingEnabled)
+                return false;
+
+            var ctorMap = new ConstructorMap(destCtor, typeMap);
+
+            foreach(var parameter in ctorParameters)
+            {
+                var resolvers = new LinkedList<MemberInfo>();
+
+                var canResolve = MapDestinationPropertyToSource(options, sourceTypeInfo, destCtor.DeclaringType, parameter.GetType(), parameter.Name, resolvers);
+                if((!canResolve && parameter.IsOptional) || CtorParamConfigurations.Any(c => c.CheckCtorParamName(parameter.Name)))
+                {
+                    canResolve = true;
+                }
+                ctorMap.AddParameter(parameter, resolvers.ToArray(), canResolve);
+            }
+
+            typeMap.ConstructorMap = ctorMap;
+
+            return ctorMap.CanResolve;
+        }
+
+        private bool MapDestinationPropertyToSource(ProfileMap options, TypeDetails sourceTypeInfo, Type destType, Type destMemberType, string destMemberInfo, LinkedList<MemberInfo> members)
```

Comment:
That seems just copy pasted from TypeMapFactory.

Reply:
It is, as it is needed in both locations.  I wasn't sure what your policy on code sharing is.

I could remove it from both locations and add it as an instance method of `ProfileMap`, as that is shared between both locations?  (I would also move `MapDestinationCtorToSource` to `ProfileMap`, too, as they are obviously linked).

Thoughts?

Reply:
OK, move them both.

Reply:
Moved